walk.go

```diff
@@ -39,19 +39,24 @@ func walk(fileSystem LstatReadDirer, path string, walkFn filepath.WalkFunc, info
 	}
 	dirEntries, err := fileSystem.ReadDir(path)
 	if err != nil {
-		return err
+		if err := walkFn(path, info, err); err != nil {
+			return err
+		}
 	}
 	sort.Sort(dirEntriesByName(dirEntries))
 	for _, dirEntry := range dirEntries {
 		name := dirEntry.Name()
 		if name == "." || name == ".." {
 			continue
 		}
+		path := filepath.Join(path, dirEntry.Name())
 		info, err := dirEntry.Info()
 		if err != nil {
-			return err
+			if err := walkFn(path, info, err); err != nil {
+				return err
```

Comment:
It feels like we should be checking `errors.Is(err, fs.SkipDir)` here like we do at line 37.

Something like:

```
if err := walkFn(path, info, err); err != nil {
  if errors.Is(err, fs.SkipDir) {
    continue
  }

  return err
}
```

Reply:
Thanks, done, great catch!